File: backend/app/schemas/leader_evaluation.py
from pydantic import BaseModel


class ScoreBreakdown(BaseModel):
    intent_score: float | None
    concept_score: float | None
    routing_score: float | None
    tool_success_score: float | None
    evidence_score: float | None
    final_score: float | None


class LeaderEvaluationListItemResponse(BaseModel):
    request_id: str
    user_question: str | None
    detected_intent: str | None
    intent_match_yn: bool | None
    detected_concepts: list[str]
    direct_concepts: list[str]
    expanded_concepts: list[str]
    unrouted_concepts: list[str]
    selected_agents: list[str]
    leader_confidence_score: float | None
    avg_evidence_confidence: float | None
    avg_data_quality_score: float | None
    avg_intent_relevance_score: float | None
    hallucination_yn: bool | None
    overall_result: str | None
    review_reason: str | None
    score_breakdown: ScoreBreakdown | None
    reviewer: str | None
    reviewed_at: str | None
    tool_success_count: int
    tool_failed_count: int
    created_at: str | None


class LeaderEvaluationListResponse(BaseModel):
    items: list[LeaderEvaluationListItemResponse]
    total: int


class LeaderDecisionResponse(BaseModel):
    detected_intent: str | None
    expected_intent: str | None
    intent_match_yn: bool | None
    detected_concepts: list[str]
    direct_concepts: list[str]
    expanded_concepts: list[str]
    unrouted_concepts: list[str]
    expected_concepts: list[str]
    missing_concepts: list[str]
    extra_concepts: list[str]
    selected_agents: list[str]
    expected_agents: list[str]
    missing_agents: list[str]
    extra_agents: list[str]
    step_count: int
    successful_step_count: int
    failed_step_count: int
    leader_confidence_score: float | None
    intent_keywords: list[str]
    intent_urgency: str | None
    memory_turns: int
    ltm_turns: int
    review_reason: str | None
    score_breakdown: ScoreBreakdown | None


class EvidenceSummaryResponse(BaseModel):
    evidence_count: int
    avg_evidence_confidence: float | None
    avg_data_quality_score: float | None
    avg_intent_relevance_score: float | None
    missing_concepts: list[str]
    grounding_score: float | None
    review_flags: list[str]


class LeaderEvaluationReviewResponse(BaseModel):
    final_answer_review: str | None
    hallucination_yn: bool | None
    overall_result: str | None
    review_comment: str | None
    reviewer: str | None
    reviewed_at: str | None


class LeaderEvaluationTraceEventResponse(BaseModel):
    id: int
    event_type: str
    agent_id: str | None
    tool_id: str | None
    status: str
    input_data: dict | None
    output_data: dict | None
    duration_ms: int | None
    created_at: str


class LeaderEvaluationEvidenceResponse(BaseModel):
    id: int
    concept_id: str | None
    source_id: str | None
    agent_id: str | None
    confidence_score: float | None
    data_quality_score: float | None
    intent_relevance_score: float | None
    response_latency_ms: int | None
    item_count: int | None
    quality_flags: dict | None
    related_evidence_ids: list[int] | None
    content_preview: dict | None
    created_at: str


class LeaderEvaluationDetailResponse(BaseModel):
    request_id: str
    user_question: str | None
    answer: str | None
    leader_decision: LeaderDecisionResponse
    evidence_summary: EvidenceSummaryResponse
    review: LeaderEvaluationReviewResponse
    trace_events: list[LeaderEvaluationTraceEventResponse]
    evidences: list[LeaderEvaluationEvidenceResponse]

File: frontend/src/types/leaderEvaluation.ts
export type ScoreBreakdown = {
  intent_score: number | null;
  concept_score: number | null;
  routing_score: number | null;
  tool_success_score: number | null;
  evidence_score: number | null;
  final_score: number | null;
};

export type LeaderEvaluationListItem = {
  request_id: string;
  user_question: string | null;
  detected_intent: string | null;
  intent_match_yn: boolean | null;
  detected_concepts: string[];
  direct_concepts: string[];
  expanded_concepts: string[];
  unrouted_concepts: string[];
  selected_agents: string[];
  leader_confidence_score: number | null;
  avg_evidence_confidence: number | null;
  avg_data_quality_score: number | null;
  avg_intent_relevance_score: number | null;
  hallucination_yn: boolean | null;
  overall_result: string | null;
  review_reason: string | null;
  score_breakdown: ScoreBreakdown | null;
  reviewer: string | null;
  reviewed_at: string | null;
  tool_success_count: number;
  tool_failed_count: number;
  created_at: string | null;
};

export type LeaderEvaluationListResponse = {
  items: LeaderEvaluationListItem[];
  total: number;
};

export type LeaderDecisionDetail = {
  detected_intent: string | null;
  expected_intent: string | null;
  intent_match_yn: boolean | null;
  detected_concepts: string[];
  direct_concepts: string[];
  expanded_concepts: string[];
  unrouted_concepts: string[];
  expected_concepts: string[];
  missing_concepts: string[];
  extra_concepts: string[];
  selected_agents: string[];
  expected_agents: string[];
  missing_agents: string[];
  extra_agents: string[];
  step_count: number;
  successful_step_count: number;
  failed_step_count: number;
  leader_confidence_score: number | null;
  intent_keywords: string[];
  intent_urgency: string | null;
  memory_turns: number;
  ltm_turns: number;
  review_reason: string | null;
  score_breakdown: ScoreBreakdown | null;
};

export type LeaderEvaluationEvidenceSummary = {
  evidence_count: number;
  avg_evidence_confidence: number | null;
  avg_data_quality_score: number | null;
  avg_intent_relevance_score: number | null;
  missing_concepts: string[];
  grounding_score: number | null;
  review_flags: string[];
};

export type LeaderEvaluationReview = {
  final_answer_review: string | null;
  hallucination_yn: boolean | null;
  overall_result: string | null;
  review_comment: string | null;
  reviewer: string | null;
  reviewed_at: string | null;
};

export type LeaderEvaluationTraceEvent = {
  id: number;
  event_type: string;
  agent_id: string | null;
  tool_id: string | null;
  status: string;
  input_data: Record<string, unknown> | null;
  output_data: Record<string, unknown> | null;
  duration_ms: number | null;
  created_at: string;
};

export type LeaderEvaluationEvidence = {
  id: number;
  concept_id: string | null;
  source_id: string | null;
  agent_id: string | null;
  confidence_score: number | null;
  data_quality_score: number | null;
  intent_relevance_score: number | null;
  response_latency_ms: number | null;
  item_count: number | null;
  quality_flags: Record<string, unknown> | null;
  related_evidence_ids: number[] | null;
  content_preview: Record<string, unknown> | null;
  created_at: string;
};

export type LeaderEvaluationDetail = {
  request_id: string;
  user_question: string | null;
  answer: string | null;
  leader_decision: LeaderDecisionDetail;
  evidence_summary: LeaderEvaluationEvidenceSummary;
  review: LeaderEvaluationReview;
  trace_events: LeaderEvaluationTraceEvent[];
  evidences: LeaderEvaluationEvidence[];
};
